src/Instrumentation/Psr15/src/Psr15Instrumentation.php

```diff
@@ -68,9 +70,7 @@ public static function register(): void
             'handle',
             pre: static function (RequestHandlerInterface $handler, array $params, string $class, string $function, ?string $filename, ?int $lineno) use ($instrumentation) {
                 $request = ($params[0] instanceof ServerRequestInterface) ? $params[0] : null;
-                $root = $request
-                    ? $request->getAttribute(SpanInterface::class)
-                    : Span::getCurrent();
+                $root = Context::getCurrent()->get(Psr15Instrumentation::$rootSpanExists);
```

Comment:
this is currently getting a boolean, and storing it in a variable that looks like it should hold a span

Reply:
ok, changed that